integration-tests/gradle/src/main/resources/add-extension-multi-module-kts/build.gradle.kts

```diff
@@ -1,10 +1,10 @@
 buildscript {
 
     repositories {
-        if (System.getProperties().containsKey("maven.repo.local")) {
```

Comment:
I think we still need this condition, when building a release, a new local repository is used. And we used to face error in that case.

Reply:
That's strange. The gradle docs states that `maven.repo.local` is taken into account automatically in `mavenLocal()` ([source](https://docs.gradle.org/current/dsl/org.gradle.api.artifacts.dsl.RepositoryHandler.html#org.gradle.api.artifacts.dsl.RepositoryHandler:mavenLocal()))

>  The location for the repository is determined as follows (in order of precedence):
>
> *    **The value of system property 'maven.repo.local' if set;**
> *    The value of element <localRepository> of ~/.m2/settings.xml if this file exists and element is set;
> *    The value of element <localRepository> of $M2_HOME/conf/settings.xml (where $M2_HOME is the value of the environment variable with that name) if this file exists and element is set;
> *    The path ~/.m2/repository.

I just checked older docs and it's been that way forever, since at least 4.0.

Are you sure there wasn't another problem that was since solved? E.g. I see you had to add code in multiple places to propagate the system property from one build to another (from Maven to Gradle, but also sometimes from one Gradle build to another).

I just tested locally with the following command executed from `integration-tests/gradle/target/classes/conditional-dependencies`:

```
~/workspaces/main/quarkus/integration-tests/gradle/gradlew -Dmaven.repo.local=$HOME/.m2/repository --refresh-dependencies --no-build-cache -Dorg.gradle.console=plain --stacktrace --info :ext-a:runtime:publishToMavenLocal :ext-a:deployment:publishToMavenLocal :ext-b:runtime:publishToMavenLocal :ext-b:deployment:publishToMavenLocal :ext-c:runtime:publishToMavenLocal :ext-c:deployment:publishToMavenLocal :ext-d:runtime:publishToMavenLocal :ext-d:deployment:publishToMavenLocal :ext-e:runtime:publishToMavenLocal :ext-e:deployment:publishToMavenLocal :ext-f:runtime:publishToMavenLocal :ext-f:deployment:publishToMavenLocal :ext-g:runtime:publishToMavenLocal :ext-g:deployment:publishToMavenLocal :ext-h:runtime:publishToMavenLocal :ext-h:deployment:publishToMavenLocal :ext-i:runtime:publishToMavenLocal :ext-i:deployment:publishToMavenLocal :ext-j:runtime:publishToMavenLocal :ext-j:deployment:publishToMavenLocal :ext-k:runtime:publishToMavenLocal :ext-k:deployment:publishToMavenLocal :ext-l:runtime:publishToMavenLocal :ext-l:deployment:publishToMavenLocal :ext-m:runtime:publishToMavenLocal :ext-m:deployment:publishToMavenLocal :ext-n:runtime:publishToMavenLocal :ext-n:deployment:publishToMavenLocal :ext-o:runtime:publishToMavenLocal :ext-o:deployment:publishToMavenLocal :ext-p:runtime:publishToMavenLocal :ext-p:deployment:publishToMavenLocal :ext-r:runtime:publishToMavenLocal :ext-r:deployment:publishToMavenLocal :ext-s:runtime:publishToMavenLocal :ext-s:deployment:publishToMavenLocal :ext-t:runtime:publishToMavenLocal :ext-t:deployment:publishToMavenLocal :ext-u:runtime:publishToMavenLocal :ext-u:deployment:publishToMavenLocal
```

As is, it works, but if I change `maven.repo.local` to some non-existing repository, e.g. `$HOME/.m2/repository3`, it fails because it doesn't find Quarkus SNAPSHOT artifacts. So, evidently, the system property `maven.repo.local` **is** taken into account...

Maybe let's try? @gsmet can reach out to me very quickly if it causes a problem during a release, and he won't hesitate to put me to work :)

Reply:
I think the CI release job we have set up should catch this (probably).

Also, we aren't running the tests anymore during the release process.

Reply:
ok thanks